app/models/ems_refresh/save_inventory.rb

```diff
@@ -36,10 +36,14 @@ def save_vms_inventory(ems, hashes, target = nil, disconnect = true)
     target = ems if target.nil? && disconnect
     log_header = "EMS: [#{ems.name}], id: [#{ems.id}]"
 
-    disconnects = if target.kind_of?(ExtManagementSystem) || target.kind_of?(Host)
-                    target.vms_and_templates.reload.to_a
-                  elsif target.kind_of?(Vm)
-                    [target.ruby_clone]
+    disconnects = if disconnect
+                    if target.kind_of?(ExtManagementSystem) || target.kind_of?(Host)
+                      target.vms_and_templates.reload.to_a
+                    elsif target.kind_of?(Vm)
+                      [target.ruby_clone]
+                    else
+                      []
+                    end
                   else
                     []
```

Comment:
I wonder if we just `disconnects.to_a.index` down here https://github.com/pkliczewski/manageiq/blob/3ced29d675318fbbe889b25b51aa0c75d06a29a1/app/models/ems_refresh/save_inventory.rb#L70 and then we don't need the two `[]` cases?

Reply:
The solution your are proposing won't work due to:

[----] E, [2017-09-04T11:47:32.216615 #5694:d81120] ERROR -- : MIQ(MiqQueue#deliver) Message id: [6897], Error: [undefined method `values' for 0:Fixnum]
[----] E, [2017-09-04T11:47:32.216776 #5694:d81120] ERROR -- : [NoMethodError]: undefined method `values' for 0:Fixnum  Method:[block in method_missing]
[----] E, [2017-09-04T11:47:32.216896 #5694:d81120] ERROR -- : /home/pkliczewski/git/manageiq/app/models/ems_refresh/save_inventory.rb:170:in `save_vms_inventory'

It looks like all the solutions are not nice. I will go with simplest approach with a comment.